WcaOnRails/app/controllers/api/v0/api_controller.rb

```diff
@@ -9,9 +10,13 @@ def bad_api_parameter(e)
 
   DEFAULT_API_RESULT_LIMIT = 20
 
+  def current_resource_owner
```

Comment:
I honestly didn't think about the name of this method because I copied it from the doorkeeper readme. Now that we're talking about it, I like `current_api_user` or even `doorkeeper_user`. Let me know which you prefer and I'll do it!

As for avoiding repeatedly hitting the database, how about doing the `@foo ||=` memoization pattern?

```ruby
def doorkeeper_user
  @doorkeeper_user ||= doorkeeper_token ? User.find_by_id(doorkeeper_token.resource_owner_id) : nil
end
```